src/core.jl

```diff
@@ -165,3 +165,10 @@ function size{SA <: StaticArray}(::Type{SA})
             SMatrix{3,3}(m) # correct - size is inferrable
         """)
 end
+
+function Base.promote_rule{SA1<:StaticArray, SA2<:StaticArray}(::Type{SA1}, ::Type{SA2})
+    T = promote_type(eltype(SA1), eltype(SA2))
```

Comment:
This is fixed on this branch; that's the whole point of the method just below this:
```julia
julia> [SVector(1,2,3), SVector(1,2)]
ERROR: no promotion exists for StaticArrays.SVector{3,Int64} and StaticArrays.SVector{2,Int64}
 in _promote_rule(::Type{StaticArrays.SVector{3,Int64}}, ::Type{StaticArrays.SVector{2,Int64}}) at /home/tim/.julia/v0.5/StaticArrays/src/core.jl:174
 in promote_type at ./promotion.jl:140 [inlined]
 in promote_typeof(::StaticArrays.SVector{3,Int64}, ::StaticArrays.SVector{2,Int64}, ::Vararg{StaticArrays.SVector{2,Int64},N}) at ./promotion.jl:156
 in vect(::StaticArrays.SVector{3,Int64}, ::Vararg{Any,N}) at ./abstractarray.jl:24
```
It's slightly counterintuitive, but it turns out to be much easier to check that the rest of the parameters are the same after you've harmonized the element type.

This won't catch things like `promote_type(SArray{(2,),Float32,1,2}, SVector{2,Int})`, but that's a separate issue.